content/200-concepts/100-components/02-prisma-client/056-composite-types.mdx

```diff
@@ -90,6 +86,47 @@ There are currently some limitations when using composite types in the Prisma Cl
 - [`findUnique`](/reference/api-reference/prisma-client-reference#findunique) can't filter on composite types
 - [`aggregrate`](/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate), [`groupBy`](/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by), [`count`](/concepts/components/prisma-client/aggregation-grouping-summarizing#count) don’t support composite operations
 
+## Default values in required composite types
+
+From v4.0.0, if you carry out a database read on a composite type when all of the following conditions are true, then Prisma Client inserts the default value into the result.
+
+Conditions:
+
+- The composite type is [required](/concepts/components/prisma-schema/data-model#optional-and-mandatory-fields), and
```

Comment:
I don't think this true, it can also provide defaults if the composite type is optional and the value is undefined.

Reply:
OK thanks, I'll amend the text

Reply:
@matthewmueller According to the description [here](https://www.notion.so/prismaio/Composite-read-default-on-undefined-1688123bac0e4d6db8da25241e0cf4d8), and the [issue description](https://github.com/prisma/prisma/issues/12665), the composite type does have to be required. At least if I understand this correctly.

Reply:
Sorry for the late review. Andrew is correct, only required fields will be filled with default values in case they're optional. Otherwise, we might end up filling values for optional fields when they're intended to be null/undefined by the user

Reply:
That makes sense and sorry for the confusion!!